roadmap/implementors-guide/validity-subsystems-overview.md

```diff
@@ -0,0 +1,113 @@
+
+### Inbound Messages
+
+- **Candidate Selection**: a candidate which satisfies system properties
+- **Candidate Validation**: a particular candidate is valid, or not
+
+### Outbound Messages
+
+- **Candidate Validation**: is this candidate in fact valid?
+- **Statement Distribution**: I testify that this candidate is (in-)valid
+
+## Candidate Validation
+
+The candidate validation subsystem handles the details of validating a candidate. For any given parachain candidate, it arranges the appropriate context: the validation function, the relay parent head data, the in-progress relay data, etc.
+
+### Inbound Messages
+
+Note that the inbound message will only actually include the PoV when we are deciding whether or not to second a particular parachain candidate. In all other cases, we need to fetch it, because we only actually have the candidate receipt.
+
+The candidate validation subsystem is required to keep a cache of parablock hashes to validation results; it should only ever actually perform validation for a given candidate once, no matter how many times the validity for that block is queried. When validation is performed, if the PoV is not included in the request, a new message will be dispatched to the overseer to fetch the appropriate PoV from the network.
+
+To prevent unbounded growth, when the candidate validation subsystem is notified by the overseer to stop work on a particular block, it should clear the cache of results pertaining to that block.
+
+- **Various Subsystems**: requests to validate a particular parablock candidate
+- **Overseer**: response to PoV request
+
+### Outbound Messages
+
+Responses from the Candidate Validation subsystem always return to the requester
+
+- **Various Subsystems**: a parablock candidate is valid, or not
+- **Overseer**: request the PoV for a candidiate with a particular hash
+
+## Statement Distribution
+
+The statement distribution subsystem sends statements to peer nodes, detects double-voting, and tabulates when a sufficient portion of the validator set has unanimously judged a candidate. When judgment is not unanimous, it escalates the issue to misbehavior arbitration.
+
+### Inbound Messages
+
+- **Overseer**: a peer node has seconded a candidate
+
+### Outbound Messages
+
+- **Candidate Validation**: double-check this peer's seconded candidate
+- **Peer validators**: Here's what I think about a candidate's validity
+- **Misbehavior Arbitration**: I disagree with a peer node about this candidate's validity
+- **Overseer**: a unanimous quorum of nodes has agreed about this candidate's validity
+
+## Misbehavior Arbitration
+
+The misbehavior arbitration subsystem kicks in when two validator nodes disagree about a candidate's validity. In this case, _all_ validators, not just those assigned to its parachain, weigh in on the validity of this candidate. The minority is slashed.
```

Comment:
There seem to be two disparate cases covered here.

1. Arbitration of blocks that are _included_ in the chain, where disputes are signaled by the chain. These blocks are guaranteed to be available, and arbitration is handled by having validators self-select based on a random process to do further approval checks. This is handled on the runtime side by the `Validity` module.
1. Arbitration of blocks that have not been included in the chain. My PR to the validity module #1176 discusses the possibility of "remote" disputes, for blocks that haven't been included in this chain. We could allow that any disagreement becomes a remote dispute, but so far I've only considered the possibility of remote disputes where a candidate is potentially backed (i.e. has a majority of support from its validator group).

Either way, I'd classify both of these as beyond scope for this PR and suggest that we not define what to do with conflicting votes at the backing stage of the pipeline and revisit those definitions in its own PR, along with the rest of the self-selection logic.

Reply:
I didn't intend to include case 1 in this; the cases I've been thinking of are all for blocks which have not yet been included. That said, I'm fine leaving the behavior of this subsystem mostly unspecified for now pending future work.